qiskit/quantum_info/operators/unitary.py

```diff
@@ -0,0 +1,268 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+# pylint: disable=invalid-name,assignment-from-no-return
+
+"""
+Tools for working with Unitary Operators.
+
+A simple unitary class and some tools.
+"""
+import copy
+import numpy
+import sympy
+from qiskit.exceptions import QiskitError
+from qiskit.circuit.gate import Gate
+from qiskit.dagcircuit import DAGCircuit
+
+
+class Unitary(Gate):
+    """Class for representing unitary operators"""
+
+    def __init__(self, matrix, *qargs, label=None, validate=True, rtol=1e-5, atol=1e-8):
+        """
+        Create unitary.
+
+        Args:
+            matrix (numpy.ndarray or list(list)): unitary matrix representation.
```

Comment:
Can this be renamed to `data` so later it can be checked for various different objects. Eg Pauli, Clifford, Channel, not just a np matrix.

I would also suggest adding a copy type init for another unitary object so if you have a unitary `u = Unitary(mat, label=l)`, then:

* `v = Unitary(u)` would make a new object with the same matrix and label
* `v = Unitary(u, label=new)` would make a new object with same matrix but different label

Reply:
sounds good

Reply:
Or maybe `obj`?